### PR TITLE
[Core] Feature pseudo-static scheme

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
This is correct

File: applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def _create_solution_scheme(self):
         elif(scheme_type == "bossak"):
             damp_factor_m = self.dynamic_settings["damp_factor_m"].GetDouble()
             mechanical_scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(damp_factor_m)
+        elif(scheme_type == "pseudo-dynamic"):
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py]: pseudo_dynamic

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py]: Ok

**Comment by philbucher** [comment]
test? :)

**Comment by loumalouomega** [comment]
I don't know how to test a pseudo-static, I was thinking to use it as a forward Euler

**Comment by loumalouomega** [comment]
Test added

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        const std::size_t this_thread = OpenMPUtils::ThisThread();
+
+        // Adding damping contribution
+        if (rD.size1() != 0 && TDenseSpace::TwoNorm(rD) > ZeroTolerance) {
+            pElement->GetFirstDerivativesVector(DerivedBaseType::mVector.v[this_thread], 0);
+            noalias(rRHSContribution) -= prod(rD, DerivedBaseType::mVector.v[this_thread]);
+        } else if (rM.size1() != 0) {
+            const double beta = rCurrentProcessInfo[mRayleighBeta];
+            pElement->GetFirstDerivativesVector(DerivedBaseType::mVector.v[this_thread], 0);
+            noalias(rRHSContribution) -= prod(rM * beta, DerivedBaseType::mVector.v[this_thread]);
```
Comment (RiccardoRossi) [inline kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h]: put multiplication by beta outside of the prod (less calculations will be done)

Comment (loumalouomega) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h]: Done

**Comment by loumalouomega** [comment]
Done

**Pull request merged**